### PR TITLE
[spanmetrics] Move spanmetrics cfg validation to ConfigValidator#Validate.

### PR DESCRIPTION
**Pull request by kovrus**
**Description:** <Describe what has changed.>

Components:
-  `spanmetrics` processor
-  `spanmetrics` connector

Move `spanmetrics` components configuration validations to `ConfigValidator#Validate`.

### CODE REVIEW

File: processor/spanmetricsprocessor/config.go [hunk 1]
```diff
@@ -79,3 +80,49 @@ func (c Config) GetAggregationTemporality() pmetric.AggregationTemporality {
 	}
 	return pmetric.AggregationTemporalityCumulative
 }
+
+// Validate checks if the processor configuration is valid
+func (c *Config) Validate() error {
```
Comment (albertteoh) [inline processor/spanmetricsprocessor/config.go:85]: nit: we're not mutating `Config`; as such I don't think we need a pointer receiver.

File: processor/spanmetricsprocessor/config.go [hunk 2]
```diff
@@ -79,3 +80,49 @@ func (c Config) GetAggregationTemporality() pmetric.AggregationTemporality {
 	}
 	return pmetric.AggregationTemporalityCumulative
 }
+
```
Comment (albertteoh) [inline processor/spanmetricsprocessor/config.go:83]: nit: What do you think about adding a compile time check for the interface implementation to make sure this doesn't break again?

```
var _ component.ConfigValidator = (Config)(nil)
```

**Review by albertteoh — APPROVED** [review]
Thanks! Just a couple of nits.

**Review by albertteoh — APPROVED** [review]

**Comment by kovrus** [comment]
cc: @Cluas

File: connector/spanmetricsconnector/config.go [hunk 3]
```diff
@@ -70,6 +74,23 @@ type Config struct {
 	Namespace string `mapstructure:"namespace"`
 }
 
+// Validate checks if the processor configuration is valid
+func (c Config) Validate() error {
+	err := validateDimensions(c.Dimensions, dropSanitizationGate.IsEnabled())
```
Comment (Cluas) [inline connector/spanmetricsconnector/config.go:79]: There will be some conflicts with this PR here, we need to watch the order. https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/18754

Comment (kovrus) [reply connector/spanmetricsconnector/config.go:79]: Yes, any of those can be merged and another one adjusted afterward.

File: processor/spanmetricsprocessor/config.go [hunk 4]
```diff
@@ -15,12 +15,16 @@
...
 import (
+	"fmt"
 	"time"
 
+	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/featuregate"
 	"go.opentelemetry.io/collector/pdata/pmetric"
 )
 
+var _ component.ConfigValidator = (*Config)(nil)
```
Comment (MovieStoreGuy) [inline processor/spanmetricsprocessor/config.go:26]: Please move to line 75 so that it appears after type definition to improve readability.

**Review by MovieStoreGuy — APPROVED** [review]
these changes seem reasonable, from what I can see it is just moving validation to be with the respective files more closely.

**Review by Cluas — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Pull request merged**